src/interface.jl

```diff
@@ -10,63 +13,56 @@
 =#
 
-Base.eltype(::Type{AbstractSciMLOperator{T}}) where T = T
-Base.eltype(::AbstractSciMLOperator{T}) where T = T
-update_coefficients!(L,u,p,t) = nothing
-update_coefficients(L,u,p,t) = L
+DEFAULT_UPDATE_FUNC(A,u,p,t) = A
+function update_coefficients!(L::AbstractSciMLOperator, u, p, t)
+    for op in getops(L)
+        update_coefficients!(op, u, p, t)
+    end
+    L
+end
 
-# Traits
-isconstant(::AbstractSciMLOperator) = false
-islinear(::AbstractSciMLOperator) = false
-issquare(::AbstractSciMLOperator) = false
-Base.iszero(::AbstractSciMLOperator) = false
-has_adjoint(L::AbstractSciMLOperator) = false # L', adjoint(L)
-has_expmv!(L::AbstractSciMLOperator) = false # expmv!(v, L, t, u)
-has_expmv(L::AbstractSciMLOperator) = false # v = exp(L, t, u)
-has_exp(L::AbstractSciMLOperator) = false # v = exp(L, t)*u
-has_mul(L::AbstractSciMLOperator) = true # du = L*u
-has_mul!(L::AbstractSciMLOperator) = false # mul!(du, L, u)
-has_ldiv(L::AbstractSciMLOperator) = false # du = L\u
-has_ldiv!(L::AbstractSciMLOperator) = false # ldiv!(du, L, u)
+for T in (
+          SciMLIdentity,
+          SciMLNullOperator,
+          SciMLScalar,
+          SciMLScaledOperator,
+          SciMLAddedOperator,
+          SciMLComposedOperator,
 
-### AbstractDiffEqLinearOperator Interface
+          SciMLMatrixOperator,
+          SciMLFactorizedOperator,
+          SciMLFunctionOperator,
+         )
 
-#=
-1. AbstractDiffEqLinearOperator <: AbstractSciMLOperator
-2. Can absorb under multiplication by a scalar. In all algorithms things like
-   dt*L show up all the time, so the linear operator must be able to absorb
-   such constants.
-4. isconstant(A) trait for whether the operator is constant or not.
-5. Optional: diagonal, symmetric, etc traits from LinearMaps.jl.
-6. Optional: exp(A). Required for simple exponential integration.
-7. Optional: expmv(A,u,p,t) = exp(t*A)*u and expmv!(v,A::DiffEqOperator,u,p,t)
-   Required for sparse-saving exponential integration.
-8. Optional: factorizations. A_ldiv_B, factorize et. al. This is only required
-   for algorithms which use the factorization of the operator (Crank-Nicholson),
-   and only for when the default linear solve is used.
-=#
+    (L::T)(u, p, t) = (update_coefficients!(L, u, p, t); L * u)
+    (L::T)(du, u, p, t) = (update_coefficients!(L, u, p, t); mul!(du, L, u))
+end
 
-# Extra standard assumptions
-isconstant(::AbstractDiffEqLinearOperator) = true
-islinear(o::AbstractDiffEqLinearOperator) = isconstant(o)
+update_coefficients!(L,u,p,t) = nothing
+update_coefficients(L,u,p,t) = L
 
-isconstant(::AbstractMatrix) = true
-islinear(::AbstractMatrix) = true
-has_adjoint(::AbstractMatrix) = true
-has_mul(::AbstractMatrix) = true
-has_mul!(::AbstractMatrix) = true
-has_ldiv(::AbstractMatrix) = true
-has_ldiv!(::AbstractMatrix) = false
-has_ldiv!(::Union{Diagonal, Factorization}) = true
+function SciMLOperator(L;
+                       update_func=DEFAULT_UPDATE_FUNC,
```

Comment:
This operator treats it as a closed set, and I don't think that's generally safe.

Reply:
rm'd